### PR TITLE
[RCL-79] Add modeling vignette.

### PR DESCRIPTION
**Pull request by patr1ckm**
This is a first version of a modeling vignette. To view, use `devtools::build_vignettes()`, and view the resulting html in `inst/docs/` in a web browser. I don't add `inst/docs/` to the repo because it will be removed later anyway.

This will fail R CMD Check because `get_metrics` is defined in #30.

Happy to hear any feedback on this, @keithing and @wlattner.

### CODE REVIEW

File: vignettes/civis_ml.Rmd [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+civis_ml(df, ...)
+civis_ml("path/to/data.csv", ...)
+civis_ml(civis_table(table_name = "schema.table", database_name = "database"), ...)
+civis_ml(civis_file(1234), ...)
+```
+
+The Brandable data is located in a Redshift table called `sample_project.premium_training_set`:
+
+```{r, eval  = FALSE}
+tab <- civis_table("sample_project.premium_training_set", "redshift-general")
```
Comment (keithing) [inline vignettes/civis_ml.Rmd]: maybe something more clear like `my_database`?  I also like the approach [here](https://github.com/civisanalytics/civis-python/blob/master/examples/CivisML.ipynb) of dynamically getting the database name.

File: vignettes/civis_ml.Rmd [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+cv_params <- list("max_depth" = c(2, 3, 5), 
+                "n_estimators" = c(50, 100, 500))
+  
+
+m <- civis_ml(tab, dependent_variable = "upgrade", 
+              model_type = "random_forest_classifier", 
+              primary_key = "brandable_user_id", 
+              excluded_columns = "residential_zip", 
+              cross_validation_parameters = cv_params)
+saveRDS(m, file = "../inst/civis_ml_brandable.rds")
```
Comment (keithing) [inline vignettes/civis_ml.Rmd]: do we want these `saveRDS` calls here?

Comment (keithing) [reply vignettes/civis_ml.Rmd]: Ah, okay I see it is referenced below.

File: vignettes/civis_ml.Rmd [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+Documentation on the meta parameters specific to each estimator are provided in `?civis_ml_*`. For example, the regularization strength parameter `C` of `sparse_logistic` is documented in `?civis_ml_sparse_logistic`.
+
+For the brandable data, we use a `random_forest` classifier to predict the probability that a customer upgrades from free to premium services. For efficiency, we can also denote a `primary_key`, and a set of `excluded_columns` that are not included in the model:
+
+
+```{r, eval = FALSE}
+library(civis)
+tab <- civis_table("sample_project.premium_training_set", "redshift-general")
+m   <- civis_ml(tab, dependent_variable = "upgrade", 
+                model_type = "random_fest_classifier",
```
Comment (wlattner) [inline vignettes/civis_ml.Rmd]: `random_forest_classifier`

Comment (wlattner) [reply vignettes/civis_ml.Rmd]: Should we show the equivalent `civis_ml_random_forest_classifier` call?

**Comment by patr1ckm** [comment]
Lots of fixes here, hopefully some improvements. Borrowed some inspiration from Keith on the intro from `civis.r.client` modeling vignette.

File: vignettes/civis_ml.Rmd (hunk 2 above)
Comment (patr1ckm) [reply vignettes/civis_ml.Rmd]: Also, this block isn't shown

File: vignettes/civis_ml.Rmd [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+civis_ml("path/to/data.csv", ...)
+civis_ml(civis_table(table_name = "schema.table", database_name = "database"), ...)
+civis_ml(civis_file(1234), ...)
+```
+
+The Brandable data is located in a Redshift table called `sample_project.premium_training_set`:
+
+```{r, eval  = FALSE}
+tab <- civis_table(table_name = "sample_project.premium_training_set", 
+                   database_name = "redshift-general")
```
Comment (keithing) [inline vignettes/civis_ml.Rmd]: Just want to make sure my other comment on `redshift-general` isn't lost.  This database name will likely be unfamiliar to most users.

File: vignettes/civis_ml.Rmd [hunk 5]
```diff
@@ -58,11 +59,11 @@ After the data source is specified, we next choose the model type. There are 9 n
 Documentation on the meta parameters specific to each estimator are provided in `?civis_ml_*`. For example, the regularization strength parameter `C` of `sparse_logistic` is documented in `?civis_ml_sparse_logistic`.
 
 For the Brandable data, we use a `random_forest` classifier to predict the probability that a customer upgrades from free to premium services. For efficiency, we can also denote a `primary_key`, and a set of `excluded_columns` that are not included in the model:
-
+It
```
Comment (keithing) [inline vignettes/civis_ml.Rmd]: Did this mean to make it in?

Comment (patr1ckm) [reply vignettes/civis_ml.Rmd]: Fixed

**Pull request merged**